### PR TITLE
refactor: Use `async/await` in `lib/plugins/aws`.

### PR DESCRIPTION
**Pull request by ifitzsimmons**
Covered the following paths:
* Files put directly into `lib/plugins/aws`

Addresses: #8368

### CODE REVIEW

**Comment by ifitzsimmons** [comment]
@pgrzesik  Some weirdness surrounding the `aysnc` tag to one of the `listFunctions.js` hooks. It thinks I deleted a line and added another, and that the added line is not tested. Let me know how you'd like me to proceed.

**Review by pgrzesik — APPROVED** [review]
Thank you @ifitzsimmons - it looks great, the missing coverage is caused by the fact that the original line wasn't tested, so don't worry about it here.

File: lib/plugins/aws/rollback.js [hunk 1]
```diff
@@ -16,8 +16,8 @@ class AwsRollback {
     Object.assign(this, validate, setBucketName, updateStack, monitorStack);
 
     this.hooks = {
-      'before:rollback:initialize': () => BbPromise.bind(this).then(this.validate),
-      'rollback:rollback': () => {
+      'before:rollback:initialize': async () => BbPromise.bind(this).then(this.validate),
```
Comment (vladgolubev) [inline lib/plugins/aws/rollback.js:19]: I wonder, what's the point of putting `async/await` without refactoring promise chains?

Comment (pgrzesik) [reply lib/plugins/aws/rollback.js:19]: To reduce the scope - it's already quite a lot of functions to be refactored and these types of refactoring are error-prone. During the next step, we will be getting rid of promise chains as well, please see the description of the issue: https://github.com/serverless/serverless/issues/8368 and corresponding ones: https://github.com/serverless/serverless/issues/8369 and https://github.com/serverless/serverless/issues/7171

Comment (coyoteecd) [reply lib/plugins/aws/rollback.js:19]: @vladgolubev @pgrzesik just to reinforce the "error-prone" point, such a simple change (the one in request function above) broke one of the plugins we're using, see https://github.com/arabold/serverless-export-env/issues/26. So yeah, it's good to make it in steps.

Comment (pgrzesik) [reply lib/plugins/aws/rollback.js:19]: Ah, right @coyoteecd, that's an unfortunate side-effect of these migrations, but hopefully it's not going to impact too many plugins. I''ll bring up the topic so we can make sure to better announce such refactors/potential breaking changes to plugins in the future

**Pull request merged**